### DISCUSSION TITLE
How to use download?

### DISCUSSION THREAD

**Discussion by dynabler** [philippta/flyscrape#45] [Q&A] [answered]
The examples for download (examples/download.js) show the usage of download INSIDE a el.map. How do you use it outside a el.map?

```
export default function({ doc }) {
  const name = doc.find('h1');
  const thumb = doc.find('img');

    return {
      name: name.text(),
      thumb: thumb.attr('src')
    }
}
```

**Comment by philippta** [comment, marked as answer]
You should be able to call the download anywhere in the  extraction function.

```javascript
export default function({ doc, absoluteURL }) {
  const name = doc.find('h1');
  const thumb = doc.find('img');

  const thumburl = absoluteURL(thumb.attr('src'));

  download(thumburl)

  return {
    name: name.text(),
    thumb: thumb.attr('src')
  }
}
```

**Comment by dynabler** [comment]
After a bit of fiddling, I got it to work. So, if I'm understanding it correct, in order to download you need 3 things:
1. `import { download } from "flyscrape/http";` at the beginning of script.js
2. absoluteURL of the image
3. a directory OR filename.extension

Challenge no. 1: directory OR filename.extension
I can't get both to work at the same time. So this doesn't work `download(thumburl, dir/name.jpg)`, since `dir` is a string and name is an earlier declared const. And because each jpg should have it's own name.

I can, however, use concat to use a dir and name for each file: `download(thumburl, "dir/" + name.text() + ".jpg")`

It works for now.

Challenge no. 2: get the path in the final JSON
The final JSON has no mention of (relative) path to the thumb. I understand the absolute path is a bit cumbersome, but the relative path should be in JSON. How do I do that?  A quick & dirty is to declare it again and use that, but it also means more points of failure. How do make a const out of `download(thumburl, "dir/" + name.text() + ".jpg")` making it less prone to errors?
